File: .kdbp/evidence/phase-9/journey.mjs
import { chromium } from 'playwright';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));
const SHOT = (name) => join(__dirname, `${name}.png`);
const URL = 'http://127.0.0.1:15179';

(async () => {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage({ viewport: { width: 1280, height: 800 } });

  // 1. Login screen
  await page.goto(URL);
  await page.waitForTimeout(500);
  await page.screenshot({ path: SHOT('01-login'), fullPage: true });
  console.log('✓ 01-login');

  // 2. Navigate to case setup via debug panel
  const debugBtn = page.locator('button', { hasText: /login/ });
  await debugBtn.click();
  await page.locator('button', { hasText: /^case$/ }).click();
  await page.waitForTimeout(300);
  await page.screenshot({ path: SHOT('02-case-setup'), fullPage: true });
  console.log('✓ 02-case-setup');

  // 3. Fill case form — use the real interactive flow
  // Type institution name
  const institutionInput = page.locator('input[placeholder*="institucion"], input[placeholder*="Institucion"], input[placeholder*="banco"], input[placeholder*="Banco"]').first();
  if (await institutionInput.count() > 0) {
    await institutionInput.fill('Banco Estado');
  }
  // Click "Credito de consumo" if visible
  const creditOption = page.locator('text=Crédito de consumo, text=Credito de consumo, text=crédito bancario').first();
  if (await creditOption.count() > 0) {
    await creditOption.click();
  }
  await page.waitForTimeout(300);
  await page.screenshot({ path: SHOT('03-case-form-filled'), fullPage: true });
  console.log('✓ 03-case-form-filled');

  // 4. Navigate to upload
  const continueBtn = page.locator('button', { hasText: /Continuar|Siguiente|Subir/ }).first();
  if (await continueBtn.count() > 0) {
    await continueBtn.click();
    await page.waitForTimeout(300);
  }
  await page.screenshot({ path: SHOT('04-upload'), fullPage: true });
  console.log('✓ 04-upload');

  // 5. Navigate to findings (analysis results) via debug panel
  const debugBtn2 = page.locator('div[style*="fixed"] button').first();
  await debugBtn2.click();
  await page.waitForTimeout(200);
  await page.locator('button', { hasText: /^findings$/ }).click();
  await page.waitForTimeout(500);
  await page.screenshot({ path: SHOT('05-findings-screen'), fullPage: true });
  console.log('✓ 05-findings-screen');

  // 6. Check if "No hay caso seleccionado" or "No hay analisis previos" is shown
  const noCase = page.locator('text=No hay caso seleccionado');
  const noAnalysis = page.locator('text=No hay analisis previos');
  if (await noCase.count() > 0) {
    console.log('  → No case selected (expected in proto mode without API)');
  } else if (await noAnalysis.count() > 0) {
    console.log('  → No analysis runs yet (expected without backend)');
  }
  await page.screenshot({ path: SHOT('06-findings-detail'), fullPage: true });
  console.log('✓ 06-findings-detail');

  // 7. Navigate to coach (prototype findings view) via debug panel
  const debugBtn3 = page.locator('div[style*="fixed"] button').first();
  await debugBtn3.click();
  await page.waitForTimeout(200);
  await page.locator('button', { hasText: /^coach$/ }).click();
  await page.waitForTimeout(500);
  await page.screenshot({ path: SHOT('07-coach-view'), fullPage: true });
  console.log('✓ 07-coach-view');

  // 8. Navigate to email view
  const debugBtn4 = page.locator('div[style*="fixed"] button').first();
  await debugBtn4.click();
  await page.waitForTimeout(200);
  await page.locator('button', { hasText: /^email$/ }).click();
  await page.waitForTimeout(500);
  await page.screenshot({ path: SHOT('08-email-view'), fullPage: true });
  console.log('✓ 08-email-view');

  await browser.close();
  console.log('\nAll screenshots saved to .kdbp/evidence/phase-9/');
})();

File: .kdbp/evidence/phase-9/journey.py
"""Phase 9 runtime journey evidence: browser smoke test of the analysis UI."""
from pathlib import Path
from playwright.sync_api import sync_playwright

DIR = Path(__file__).parent
SHOT = lambda name: str(DIR / f"{name}.png")
URL = "http://127.0.0.1:15179"


def run():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page(viewport={"width": 1280, "height": 800})

        # 1. Login screen
        page.goto(URL)
        page.wait_for_timeout(600)
        page.screenshot(path=SHOT("01-login"), full_page=True)
        print("ok 01-login")

        # 2. Open debug panel and go to case setup
        page.locator("div[style*='fixed'] button").first.click()
        page.wait_for_timeout(200)
        page.locator("button", has_text="case").click()
        page.wait_for_timeout(400)
        page.screenshot(path=SHOT("02-case-setup"), full_page=True)
        print("ok 02-case-setup")

        # 3. Navigate to upload via debug panel
        page.locator("div[style*='fixed'] button").first.click()
        page.wait_for_timeout(200)
        page.locator("button", has_text="upload").click()
        page.wait_for_timeout(400)
        page.screenshot(path=SHOT("03-upload"), full_page=True)
        print("ok 03-upload")

        # 4. Navigate to findings (AnalysisResults screen)
        page.locator("div[style*='fixed'] button").first.click()
        page.wait_for_timeout(200)
        page.locator("button", has_text="findings").click()
        page.wait_for_timeout(600)
        page.screenshot(path=SHOT("04-findings-screen"), full_page=True)
        print("ok 04-findings-screen")

        # 5. Check UI states
        if page.locator("text=No hay caso seleccionado").count() > 0:
            print("   -> No case selected (expected: no caseId in nav state)")
        elif page.locator("text=No hay analisis previos").count() > 0:
            print("   -> No analysis runs (expected: no backend)")

        # 6. Navigate to coach (prototype findings comparison)
        page.locator("div[style*='fixed'] button").first.click()
        page.wait_for_timeout(200)
        page.locator("button", has_text="coach").click()
        page.wait_for_timeout(500)
        page.screenshot(path=SHOT("05-coach-prototype"), full_page=True)
        print("ok 05-coach-prototype")

        # 7. Navigate to email
        page.locator("div[style*='fixed'] button").first.click()
        page.wait_for_timeout(200)
        page.locator("button", has_text="email").click()
        page.wait_for_timeout(500)
        page.screenshot(path=SHOT("06-email"), full_page=True)
        print("ok 06-email")

        # 8. Back to findings to verify the screen renders consistently
        page.locator("div[style*='fixed'] button").first.click()
        page.wait_for_timeout(200)
        page.locator("button", has_text="findings").click()
        page.wait_for_timeout(500)
        page.screenshot(path=SHOT("07-findings-revisit"), full_page=True)
        print("ok 07-findings-revisit")

        browser.close()
        print("\nAll screenshots saved to .kdbp/evidence/phase-9/")


if __name__ == "__main__":
    run()
